test/unit.coffee

```diff
@@ -0,0 +1,100 @@
+assert = require "assert"
+mocha = require "mocha"
+_ = require "underscore"
+thriftPool = {_private} = require "../lib/index"
+async  = require 'async'
+sinon = require "sinon"
+{EventEmitter} = require "events"
+
+connection_mock = ->
+  connection = new EventEmitter()
+  connection.connection =
+    setKeepAlive: sinon.stub()
+    end: sinon.stub()
+  connection
+
+describe "thrift-pool", ->
+  before ->
+    @mock_connection = connection_mock()
+    @thriftService =
+      Client: class
+        fn: ->
+        fn2: ->
+    @initialized_thrift_service =
+      fn: sinon.stub().yields null, "xyz"
+      fn2: sinon.stub().yields new Error("error"), null
+    @thrift =
+      createConnection: () =>
+        @mock_connection.emit "connect"
+        @mock_connection
+      createClient: sinon.stub().returns @initialized_thrift_service
+    @wrappedPool = thriftPool @thrift, @thriftService, {"host", "port"}
+
+  it "returns an object with all the original keys of the thrift service", ->
+    assert @wrappedPool.fn
+    assert.equal typeof @wrappedPool.fn, "function"
+    assert.deepEqual _(@wrappedPool).keys(), ["fn", "fn2"]
+
+  it "acquires a connection from the pool when calling a method", (done) ->
+    @wrappedPool.fn "foo", "bar", (err, data) =>
+      assert @initialized_thrift_service.fn.calledWith "foo", "bar"
+      assert.equal @thrift.createClient.args[0][0], @thriftService
+      assert.equal @thrift.createClient.args[0][1], @mock_connection
+      done()
+
+  it 'returns same results that service client would', (done) ->
+    async.series [
+      (cb) =>
+        @wrappedPool.fn "foo", (err, data) =>
+          assert.equal data, "xyz"
+          cb()
+      (cb) =>
+        @wrappedPool.fn2 "foo", (err, data) =>
+          assert.notEqual err, null
+          assert.equal null, data
+          cb()
+    ], () ->
```

Comment:
parens removed: 8aea12564a952c786cd452abb27be57450bf200c
linting added: a7e53ce1dd794c368a216d21eb451aae9edcedcf

Note:  This [PR](https://github.com/Clever/coffeescript-style-guide/tree/publish-public-styleguide) on coffeescript-style-guide is trying to make coffeescript-style-guide public.  Currently, build on drone fails because npm install fails.